backend/main/src/Group/UseCases/UpdateFoodOfContainerUseCase/UpdateFoodOfContainerUseCase.ts

```diff
@@ -0,0 +1,126 @@
+import { IContainerRepository } from "src/Group/Domain/IContainerRepository";
+import { ContainerId } from "src/Group/Domain/Entities/Container";
+import { IUseCase } from "src/Shared";
+import { Err, Ok, Result } from "result-ts-type";
+import {
+	UpdateFoodOfContainerUseCaseErrorType,
+	IUpdateFoodOfContainerUseCase,
+	ContainerIsNotExisting,
+	UserIsNotAuthorized,
+	GroupIsNotExisting,
+} from "src/Group/UseCases/UpdateFoodOfContainerUseCase/IUpdateFoodOfContainerUseCase";
+import { IGroupRepository } from "src/Group/Domain/IGroupRepository";
+import { UserId } from "src/User";
+import { Quantity } from "src/Group/Domain/ValueObjects/Quantity";
+import { Food, FoodId, IFoodProps } from "src/Group/Domain/Entities/Food";
+import { Expiry } from "src/Group/Domain/ValueObjects/Expiry";
+import { Unit } from "src/Group/Domain/ValueObjects/Unit";
+
+/**
+ * Update Food of a container.
+ */
+export class UpdateFoodOfContainerUseCase
+	implements
+		IUseCase<
+			IUpdateFoodOfContainerUseCase,
+			undefined,
+			UpdateFoodOfContainerUseCaseErrorType
+		>
+{
+	private readonly containerRepository: IContainerRepository;
+	private readonly groupRepository: IGroupRepository;
+
+	constructor(
+		containerRepository: IContainerRepository,
+		groupRepository: IGroupRepository,
+	) {
+		this.containerRepository = containerRepository;
+		this.groupRepository = groupRepository;
+	}
+
+	public async execute(
+		request: IUpdateFoodOfContainerUseCase,
+	): Promise<Result<undefined, UpdateFoodOfContainerUseCaseErrorType>> {
+		const { userId, containerId, foodId, name, unit, quantity, expiry } =
+			request;
+		const containerIdOrError = ContainerId.create(containerId);
+		if (!containerIdOrError.ok) {
+			return Err(containerIdOrError.error);
+		}
+		const containerIdValue = containerIdOrError.value;
+
+		// check the user is the member of the group
+		const group = await this.groupRepository.find(containerIdValue);
```

Comment:
You can get the container at the same time as getting the group concurrently using Promise.all(). It is time-effective.

```suggestion
		const [group, container] = await Promise.all([ 
		this.groupRepository.find(containerIdValue),
		this.containerRepository.find(containerIdValue)
		]);
```

Reply:
Thank you! I'll change similar code in other usecase files.